### PR TITLE
NEW Import PDF export functionality from BasePage in cwp/cwp

### PR DESCRIPTION
**Pull request by robbieaverill**
Issue: https://github.com/silverstripe/cwp/issues/31

Requires https://github.com/silverstripe/cwp/pull/32

### CODE REVIEW

**Comment by robbieaverill** [comment]
Noted issue: clicking the "Export to PDF" link for a page initially fails with this message:

```
Fatal error: wkhtmltopdf failed: Exit with code 1 due to network error: ContentNotFoundError in /webroot/vendor/cwp/cwp-pdfexport/src/Extensions/PdfExportControllerExtension.php on line 164

```

Subsequent requests succeed and open the PDF.

I suspect this could be a macOS issue with piping the wkhtmltopdf output to stdout. If we're happy, I think we should get this merge and raise that as a separate issue. We could write it to the PHP temp dir instead of piping to stdout.

File: src/Extensions/PdfExportControllerExtension.php [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        if (!$binaryPath || !is_executable($binaryPath)) {
+            if (Environment::getEnv('WKHTMLTOPDF_BINARY')
+                && is_executable(Environment::getEnv('WKHTMLTOPDF_BINARY'))
+            ) {
+                $binaryPath = Environment::getEnv('WKHTMLTOPDF_BINARY');
+            }
+        }
+
+        if (!$binaryPath) {
+            user_error('Neither WKHTMLTOPDF_BINARY nor BasePage.wkhtmltopdf_binary are defined', E_USER_ERROR);
```
Comment (NightJar) [inline src/Extensions/PdfExportControllerExtension.php]: Error message needs to no reference `BasePage` anymore I think, it'll send people off on a wild goose chase.

File: src/Extensions/PdfExportExtension.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    /**
+     * Build pdf link for template.
+     */
+    public function PdfLink()
+    {
+        if (!$this->owner->config()->get('pdf_export')) {
+            return false;
+        }
+
+        $path = $this->owner->getPdfFilename();
```
Comment (NightJar) [inline src/Extensions/PdfExportExtension.php]: owner is going the long way around for this call.

**Review by NightJar — APPROVED** [review]
Seems good, other than what appear to be some non-critical superfluous (and magic) function calls. Would you like to address them, or merge as is?

File: src/Extensions/PdfExportControllerExtension.php (hunk 1 above)
Comment (robbieaverill) [reply src/Extensions/PdfExportControllerExtension.php]: Yeah good call, will update

File: src/Extensions/PdfExportExtension.php (hunk 2 above)
Comment (robbieaverill) [reply src/Extensions/PdfExportExtension.php]: ~~I think this follows the standard practice for extensions~~ *edited* there's almost no reason to do this so may as well update it. Will push.

**Comment by robbieaverill** [comment]
I've raised an issue for the error I noted above here: https://github.com/silverstripe/cwp-pdfexport/issues/2

**Pull request merged**